src/main/java/frc/robot/end_effector/EndEffectorIOVortex.java

```diff
@@ -0,0 +1,92 @@
+        private final RelativeEncoder algaeWristEncoder = algaeWrist.getEncoder();
+        private final DigitalInput coralIntakeBeamBreakDigitalInput = new DigitalInput(EndEffectorConstants.coralIntakeBeamBreakDIO); 
+        private final static LaserCan LaserCanSensor = new LaserCan(EndEffectorConstants.AlgaeIntakeLaserCAN); 
+    public EndEffectorIOVortex () {
+
+        SparkFlexConfig config = new SparkFlexConfig();
+
+        config.idleMode(IdleMode.kBrake).smartCurrentLimit(EndEffectorConstants.coralIntakecurrentLimit).voltageCompensation(12.0);
+        config.idleMode(IdleMode.kBrake).smartCurrentLimit(EndEffectorConstants.algaeIntakecurrentLimit).voltageCompensation(12.0);
+        config.idleMode(IdleMode.kBrake).smartCurrentLimit(EndEffectorConstants.algaeWristcurrentLimit).voltageCompensation(12.0);
+
+        config.encoder
+            .positionConversionFactor(2.0 * Math.PI)
+            .velocityConversionFactor((2.0 * Math.PI) / 60.0)
+            .uvwMeasurementPeriod(10)
+            .uvwAverageDepth(2);
+
+        coralIntake.configure(config, ResetMode.kResetSafeParameters, PersistMode.kPersistParameters);
+        algaeIntake.configure(config, ResetMode.kResetSafeParameters, PersistMode.kPersistParameters);
+        algaeWrist.configure(config, ResetMode.kResetSafeParameters, PersistMode.kPersistParameters);
+        LaserCan.Measurement measurement = LaserCanSensor.getMeasurement();
+        if (measurement != null && measurement.status == LaserCanInterface.LASERCAN_STATUS_VALID_MEASUREMENT) {
+            EndEffectorConstants.laserDistance = measurement.distance_mm;
+        } else {
+            EndEffectorConstants.laserDistance = -1.0;  // Error value
+        }
+
+        }
+        
+        
+    @Override
+    public void updateInputs (EndEffectorIOInputs inputs) {
+
+        inputs.coralIntakePosition = coralIntakeEncoder.getPosition();
+        inputs.coralIntakeVelocity = coralIntakeEncoder.getVelocity();
+        inputs.coralIntakeVoltage = coralIntake.getAppliedOutput() * coralIntake.getBusVoltage();
+        inputs.coralIntakeCurrent = coralIntake.getOutputCurrent();
+
+        inputs.algaeIntakePosition = algaeIntakeEncoder.getPosition();
+        inputs.algaeIntakeVelocity = algaeIntakeEncoder.getVelocity();
+        inputs.algaeIntakeVoltage = algaeIntake.getAppliedOutput() * algaeIntake.getBusVoltage();
+        inputs.algaeIntakeCurrent = algaeIntake.getOutputCurrent();
+        
+        inputs.algaeWristPosition = algaeWristEncoder.getPosition();
+        inputs.algaeWristVelocity = algaeWristEncoder.getVelocity();
+        inputs.algaeWristVoltage = algaeWrist.getAppliedOutput() * algaeWrist.getBusVoltage();
+        inputs.algaeWristCurrent = algaeWrist.getOutputCurrent();
+
+        inputs.coralIntakeBeamBreak = coralIntakeBeamBreakDigitalInput.get();
+        inputs.laserDistance = EndEffectorConstants.laserDistance;
```

Comment:
The values in constants really shouldn't be changing at all. Instead of referring the distance from constants, you should find the distance in this function.